src/PrestaShopBundle/Entity/Repository/OrderInvoiceRepository.php

```diff
@@ -61,18 +61,15 @@ public function __construct(Connection $connection, $tablePrefix)
      */
     public function countByOrderState(array $shopIds)
     {
-        $sql = <<<'SQL'
-SELECT COUNT(o.id_order) AS nbOrders, o.current_state as id_order_state
-FROM `{table_prefix}order_invoice` oi
-LEFT JOIN `{table_prefix}orders` o ON oi.id_order = o.id_order
-WHERE o.id_shop IN(:shopIds)
-AND oi.number > 0
-GROUP BY o.current_state
-SQL;
+        $sql = 'SELECT COUNT(o.id_order) AS nbOrders, o.current_state as id_order_state
+            FROM `{table_prefix}order_invoice` oi
+            LEFT JOIN `{table_prefix}orders` o ON oi.id_order = o.id_order
+            WHERE o.id_shop IN('.implode(',', array_map('intval', $shopIds)).')
+            AND oi.number > 0
+            GROUP BY o.current_state';
         $sql = str_replace('{table_prefix}', $this->tablePrefix, $sql);
 
         $statement = $this->connection->prepare($sql);
-        $statement->bindValue('shopIds', implode(',', array_map('intval', $shopIds)));
```

Comment:
Hello, I think we can keep the `bindValue()` which has benefits by passing the right parameter type `Connection::PARAM_STR_ARRAY`

It should look like :
```
$statement->bindValue('shopIds', $shopIds, Connection::PARAM_STR_ARRAY);
```
can you try and explore this?

Reply:
Already tried with PARAM_INT_ARRAY and PARAM_STR_ARRAY but we'll end up with an exeception in both ways :

```
An exception occurred while executing 'SELECT COUNT(o.id_order) AS nbOrders, o.current_state as id_order_state
FROM `ps_order_invoice` oi
LEFT JOIN `ps_orders` o ON oi.id_order = o.id_order
WHERE o.id_shop IN(:shopIds)
AND oi.number > 0
GROUP BY o.current_state' with params [{"2":2,"1":1}]:

Notice: Array to string conversion
```

Looks like PDO / DBAL bindValue() is not array friendly - In order to work , query should be rewritten with QueryBuilder but my patch would work aswell...